Sources/_FoundationInternals/String/BidirectionalCollection.swift

```diff
@@ -12,19 +12,16 @@
 
 extension BidirectionalCollection where Index == String.Index {
     internal func _alignIndex(roundingDown i: Index) -> Index {
-        return i < endIndex ? index(before: index(after: i)) : i
+        index(i, offsetBy: 0)
```

Comment:
This is a subtle change:

1.  We know we aren't running on a stdlib predating Swift 5.7, so `index(i, offsetBy: 0)` can safely be used as a substitute for `_index(roundingDown:)` -- it does the same thing, more efficiently than stepping forward-and-back.
2. If we're at the end, this returns the rounded off value, not the original input. (On the off chance someone tried to apply an unaligned index beyond the end index.